frontend/guardian.rb

```diff
@@ -98,30 +99,14 @@ def rejected_beep(gpio_command)
     tag = $2
   end
 
-  tag_log = {
+  tag_log = TagLog.create(
     card_type: tag =~ /^c/ ? 'clipper' : 'rfid',
     card_number: tag,
     held_tag: flag == 'h' ? true : false
-  }
-
-  authorization = DoorAuthorization.first(card_number: tag)
-
-  if authorization
-    tag_log[:name] = authorization.name
-
-    unless authorization.expired? || ! authorization.active?
-      tag_log[:door_opened] = true
-    end
-  end
-
-  TagLog.create(tag_log)
+  )
 
-  if tag_log[:door_opened]
-    # open the door
-    open_door(gpio_command)
-    accepted_beep(gpio_command)
-    close_door(gpio_command)
-  else
-    rejected_beep(gpio_command)
+  if tag_log.door_opened
```

Comment:
Flow suggestion: The flow here is somewhat strange.  It was strange before the commit, but has gotten stranger now.  The authentication logic is embedded into the tag_log, where set_name_and_opened now does the authorization checks as a side effect of logging the tags in the system.  A cleaner flow might be to do the authorization checks in the guardian logic and then pass the result into the tag logging object.